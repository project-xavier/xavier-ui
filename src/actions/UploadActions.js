import { uploadFile } from '../api/upload';

export const ActionTypes = {
    UPLOAD_REQUEST: 'UPLOAD_REQUEST',
    UPLOAD_PROGRESS: 'UPLOAD_PROGRESS',
    UPLOAD_CLEAR: 'UPLOAD_CLEAR'
};

export const uploadRequest = (customerId, file, config) => {
    const formData = new FormData();
    formData.append('file', file, file.name);
    return {
        type: ActionTypes.UPLOAD_REQUEST,
        payload: uploadFile(customerId, formData, config),
        meta: {
            file,
            notifications: {
                rejected: {
                    variant: 'danger',
                    title: `Failed to upload file ${file.name}`
                }
            }
        }
    };
};

export const uploadProgress = (file, progress) => ({
    type: ActionTypes.UPLOAD_PROGRESS,
    payload: {
        file,
        progress
    }
});

export const uploadClear = () => ({
    type: ActionTypes.UPLOAD_CLEAR
});

