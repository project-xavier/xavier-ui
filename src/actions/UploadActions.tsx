import { uploadFile } from '../api/upload';
import { GenericAction } from '../models/action';
import { Upload } from '../models';

export const ActionTypes = {
  UPLOAD_REQUEST: 'UPLOAD_REQUEST',
  UPLOAD_PROGRESS: 'UPLOAD_PROGRESS',
  SELECT_UPLOAD_FILE: 'SELECT_UPLOAD_FILE',
  RESET_UPLOAD_FILE: 'RESET_UPLOAD_FILE',
};

export const uploadRequest = (upload: Upload, config = {}): GenericAction => {
  const formData = new FormData();
  Object.keys(upload).forEach((key) => formData.append(key, upload[key].toString()));

  formData.set('file', upload.file, upload.file.name);
  return {
    type: ActionTypes.UPLOAD_REQUEST,
    payload: uploadFile(formData, config),
    meta: {
      file: upload.file,
      noError: true,
      notifications: {},
    },
  };
};

export const uploadProgress = (progress: number): GenericAction => ({
  type: ActionTypes.UPLOAD_PROGRESS,
  payload: {
    progress,
  },
});

export const selectUploadFile = (file: File): GenericAction => ({
  type: ActionTypes.SELECT_UPLOAD_FILE,
  payload: {
    file,
  },
});

export const resetUploadFile = (): GenericAction => ({
  type: ActionTypes.RESET_UPLOAD_FILE,
  payload: {},
});
