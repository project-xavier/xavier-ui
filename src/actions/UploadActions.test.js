jest.mock('../api/apiClient', () => ({
    get: jest.fn(() => Promise.resolve({})),
    post: jest.fn(() => Promise.resolve({})),
    put: jest.fn(() => Promise.resolve({})),
    delete: jest.fn(() => Promise.resolve({}))
}));

import * as actionGenerator from './UploadActions';

describe('uploadRequest', () => {
    it('returns a state object', () => {
        expect(actionGenerator.uploadRequest('myCustomerId', new File([ '' ], 'myFile.zip'), {})).toMatchSnapshot();
    });
});

describe('uploadProgress', () => {
    it('returns a state object', () => {
        expect(actionGenerator.uploadProgress(new File([ '' ], 'myFile.zip'), 60)).toMatchSnapshot();
    });
});

describe('uploadClear', () => {
    it('returns a state object', () => {
        expect(actionGenerator.uploadClear()).toMatchSnapshot();
    });
});
