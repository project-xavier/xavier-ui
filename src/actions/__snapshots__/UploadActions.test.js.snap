// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uploadClear returns a state object 1`] = `
Object {
  "type": "UPLOAD_CLEAR",
}
`;

exports[`uploadProgress returns a state object 1`] = `
Object {
  "payload": Object {
    "file": File {},
    "progress": 60,
  },
  "type": "UPLOAD_PROGRESS",
}
`;

exports[`uploadRequest returns a state object 1`] = `
Object {
  "meta": Object {
    "file": File {},
    "notifications": Object {
      "rejected": Object {
        "title": "Failed to upload file myFile.zip",
        "variant": "danger",
      },
    },
  },
  "payload": Promise {},
  "type": "UPLOAD_REQUEST",
}
`;
